sdk/messaging/azservicebus/sender.go

```diff
@@ -184,14 +140,21 @@ func (sender *Sender) createSenderLink(ctx context.Context, session internal.AMQ
 	return amqpSender, nil, nil
 }
 
-func newSender(ns internal.NamespaceWithNewAMQPLinks, queueOrTopic string, cleanupOnClose func()) (*Sender, error) {
+type newSenderArgs struct {
+	ns             internal.NamespaceWithNewAMQPLinks
+	queueOrTopic   string
+	cleanupOnClose func()
+	retryOptions   RetryOptions
+}
+
+func newSender(args newSenderArgs, retryOptions RetryOptions) (*Sender, error) {
```

Comment:
Redundant `retryOptions` arg?

Reply:
Fixed this - it is redundant, but only because I shouldn't have put retryOptions in the args!